src/main/java/ch/njol/skript/expressions/ExprSets.java

```diff
@@ -118,59 +121,61 @@ public ItemStack next() {
 				public void remove() {}
 				
 			};
-		} else {
-			@SuppressWarnings("null")
-			final Iterator<ItemType> it = new ArrayIterator<>(types.getArray(e));
-			if (!it.hasNext())
-				return null;
-			iter = new Iterator<ItemStack>() {
-				
-				@SuppressWarnings("null")
-				Iterator<ItemStack> current = it.next().getAll().iterator();
-				
-				@SuppressWarnings("null")
-				@Override
-				public boolean hasNext() {
-					while (!current.hasNext() && it.hasNext()) {
-						current = it.next().getAll().iterator();
-					}
-					return current.hasNext();
-				}
-				
-				@SuppressWarnings("null")
-				@Override
-				public ItemStack next() {
-					if (!hasNext())
-						throw new NoSuchElementException();
-					return current.next();
+		Iterator<ItemType> it = new ArrayIterator<>(types.getArray(event));
+		if (!it.hasNext())
+			return null;
+		Iterator<Object> iter;
+		iter = new Iterator<Object>() {
+			
+			Iterator<ItemStack> current = it.next().getAll().iterator();
+			
+			@Override
+			public boolean hasNext() {
+				while (!current.hasNext() && it.hasNext()) {
+					current = it.next().getAll().iterator();
 				}
-				
-				@Override
-				public void remove() {}
-				
-			};
-		}
-		
-		if (!blocks)
-			return iter;
-		
-		return new CheckedIterator<>(iter, new NullableChecker<ItemStack>() {
+				return current.hasNext();
+			}
+			
+			@Override
+			public ItemStack next() {
+				if (!hasNext())
+					throw new NoSuchElementException();
+				return current.next();
+			}
+			
+			@Override
+			public void remove() {}
+			
+		};
+
+		return new CheckedIterator<Object>(iter, new NullableChecker<Object>() {
 			@Override
-			public boolean check(final @Nullable ItemStack is) {
-				return is != null && is.getType().isBlock();
+			public boolean check(@Nullable Object ob) {
+				if (ob == null)
+					return false;
+				if (ob instanceof ItemStack)
+					if (!((ItemStack) ob).getType().isBlock())
+						return false;
+				return true;
 			}
 		});
 	}
 	
 	@Override
-	public Class<? extends ItemStack> getReturnType() {
+	public Class<? extends Object> getReturnType() {
+		if (pattern == 4)
+			return Color.class;
 		return ItemStack.class;
 	}
 	
 	@Override
-	public String toString(final @Nullable Event e, final boolean debug) {
-		final Expression<ItemType> types = this.types;
-		return (blocks ? "blocks" : "items") + (types != null ? " of type" + (types.isSingle() ? "" : "s") + " " + types.toString(e, debug) : "");
+	public String toString(@Nullable Event event, boolean debug) {
+		if (event == null)
+			return "sets of data. Pattern " + pattern;
```

Comment:
Is there a reason for this?

Reply:
When you have debug mode enabled it has no event and throws an error of no check with literals.